lib/private/Config.php

```diff
@@ -226,7 +225,15 @@ private function readData() {
 			}
 		}
 
-		$this->envCache = getenv();
+		// grab any "NC_" environment variables
+		$envRaw = getenv();
+		// only save environment variables prefixed with "NC_" in the cache
+		foreach ($envRaw as $rawEnvKey => $rawEnvValue) {
+			if (str_starts_with($rawEnvKey, self::ENV_PREFIX)) {
+				$realKey = explode(self::ENV_PREFIX, $rawEnvKey)[1];
```

Comment:
substr would me faster I think
```suggestion
				$realKey = substr($rawEnvKey,0,strlen(self::ENV_PREFIX));
```
Ideally with the strlen stored in a var before the foreach.

Reply:
Good call. Thanks. Adjusted your suggestion ever so slightly (I want the second half, not the prefix itself).